listenstore/listenstore/listenstore.py

```diff
@@ -188,24 +193,70 @@ def get_playing_now(self, user_id):
+    LISTENCOUNT_CACHE_TIME_HIGH = 24 * 60 * 60 # in seconds. 1 day
+    LOW_LISTEN_THRESHOLD = 1000
+    MEDIUM_LISTEN_THRESHOLD = 10000
+
     def __init__(self, conf):
         ListenStore.__init__(self, conf)
         self.redis = Redis(conf['REDIS_HOST'])
         self.influx = InfluxDBClient(host=conf['INFLUX_HOST'], port=conf['INFLUX_PORT'], database=conf['INFLUX_DB_NAME'])
 
 
-    def _get_num_listens_for_user(self, user_name):
-        count = self.redis.get(REDIS_INFLUX_USER_LISTEN_COUNT + user_hash(user_name))
-        if count:
-            return int(count)
+    def get_listen_count_for_user(self, user_name, need_exact = False):
+        """ Returns total listen count for user with user_name passed.
 
-        results = self.influx.query("""SELECT count(*) 
-                                         FROM listen 
-                                        WHERE user_name = '%s'""" % (user_name, from_ts, to_ts, limit))
-        result = results.get_points(measurement='listen')
-#        count = self.redis.get(REDIS_INFLUX_USER_LISTEN_COUNT + user_hash(user_name))
+            Args:
+                user_name: user name of user whose listen count is to be found
+                need_exact: signifies if cached value in redis will suffice or if exact value is needed
+        """
 
-        return 0
+        if not need_exact:
+            # check if the user's listen count is already in redis
+            # if already present return it directly instead of calculating it again
+            count = self.redis.get(REDIS_INFLUX_USER_LISTEN_COUNT + user_name)
+            if count:
+                return int(count)
+
+        try:
+            results = self.influx.query("""SELECT count(*)
+                                             FROM listen
+                                            WHERE user_name = '%s'""" % (user_name))
+        except (InfluxDBServerError, InfluxDBClientError) as e:
+            self.log.error("Cannot query influx: %s" % str(e))
+            raise
+
+        # get the number of listens from the json
+        try:
+            count = results.raw['series'][0]['values'][0][1]
+        except KeyError:
+            count = 0
+
+        # put this value into redis with expiry time based on the number of listens
+        # that the user has
+        if count <= InfluxListenStore.LOW_LISTEN_THRESHOLD:
+            self.redis.setex(REDIS_INFLUX_USER_LISTEN_COUNT + user_name, count, InfluxListenStore.LISTENCOUNT_CACHE_TIME_LOW)
+        elif count <= InfluxListenStore.MEDIUM_LISTEN_THRESHOLD:
+            self.redis.setex(REDIS_INFLUX_USER_LISTEN_COUNT + user_name, count, InfluxListenStore.LISTENCOUNT_CACHE_TIME_MEDIUM)
+        else:
+            self.redis.setex(REDIS_INFLUX_USER_LISTEN_COUNT + user_name, count, InfluxListenStore.LISTENCOUNT_CACHE_TIME_HIGH)
+        return int(count)
+
+
+    def reset_listen_count(self, user_name):
+        """ Function to reset the listen count of user
```

Comment:
Documentation doesn't need to say "function to....". Just say what it does.

This method is named `reset_...`, but returns some data. Choose what it should do and change the name or return value as required